### ISSUE TITLE
VDIF output smearing

### ISSUE THREAD

**Issue by bwmeyers** [CIRA-Pulsars-and-Transients-Group/vcsbeam#45]
@cplee1 has noticed that essentially for all MSPs with small-scale structures, we are not able to resolve them (whereas in the legacy beamformer, and even in the PSRFITS output we can at least see hints of structures). 

This hints at two possibilities:

- `DSPSR`'s coherent dedispersion is not behaving as expected on our VDIF data (need to check headers, ensure consistency)
- our VDIF writing is getting something wrong such that the signals are smeared even after coherent dedispersion

It's more than likely the latter, but the first is easier to check so we should do that.

**Comment by robotopia** [comment]
Perhaps the easiest way to check the latter is to exploit the _**Power of Git**_ to see whether any of the  relevant VDIF-writing code has changed since the last time someone (Dilpreet??) used it successfully to look at find structure.

**Comment by cplee1** [comment]
Below are examples for J0034-0534 (P=1.9 ms) and J2241-5236 (P=2.2 ms). I've checked that I can see smearing when processing both legacy and MWAX observations with VCSBeam/dspsr. I've also checked the dispersion delay over the filterbank channels to rule out intrachannel smearing. The dspsr filterbanks are processed with 80 kHz fine channels, which should allow for 12.5 us time resolution.

![J2241_example](https://github.com/CIRA-Pulsars-and-Transients-Group/vcsbeam/assets/96163140/2399e984-051b-40f4-9cdf-b022496c2897)
![J0034_example](https://github.com/CIRA-Pulsars-and-Transients-Group/vcsbeam/assets/96163140/a1d62855-5919-42fb-9dbd-c58df864b90c)

**Comment by robotopia** [comment]
Possibly related to this is the fact that the VDIF output also had zeros at the front, and strange artefacts at the second boundaries. I believe I have fixed at least the zeros-out-the-front error in f0433e7 (branch "bug/ipfb_broken"). It remains to be seen whether this also fixes the other problems.
@cplee1 I'll leave it for you to check and report back.

**Comment by robotopia** [comment]
FYI: the other commits in `bug/ipfb_broken` include patches for:
- 7522679: including `NPOL 2` in the VDIF header (but this may not be the best way to do it)
- 7522679: Adding back in the `-n` option to the beamformer to chunk the processing
- 25ac38e: Changed function names in accordance with `mwa_hyperbeam`'s latest version (v0.7)

**Comment by bwmeyers** [comment]
Right, so Chris will need to merge his changes into this branch first. And
then we can merge into the main branch (after testing, of course).
